Flapping test: tests/components/upnp/test_sensor.py::test_derived_upnp_sensors

### The problem

```
[gw0] linux -- Python 3.9.0 /__w/core/core/venv/bin/python3

hass = <homeassistant.core.HomeAssistant object at 0x7f5ab9aeab20>
setup_integration = <tests.common.MockConfigEntry object at 0x7f5abbe5ca00>

    async def test_derived_upnp_sensors(
        hass: HomeAssistant, setup_integration: MockConfigEntry
    ):
        """Test derived sensors."""
        coordinator: UpnpDataUpdateCoordinator = hass.data[DOMAIN][
            setup_integration.entry_id
        ]
    
        # First poll.
        kib_s_received_state = hass.states.get("sensor.mock_name_kib_s_received")
        kib_s_sent_state = hass.states.get("sensor.mock_name_kib_s_sent")
        packets_s_received_state = hass.states.get("sensor.mock_name_packets_s_received")
        packets_s_sent_state = hass.states.get("sensor.mock_name_packets_s_sent")
        assert kib_s_received_state.state == "unknown"
        assert kib_s_sent_state.state == "unknown"
        assert packets_s_received_state.state == "unknown"
        assert packets_s_sent_state.state == "unknown"
    
        # Second poll.
        now = dt_util.utcnow()
        with patch(
            "homeassistant.components.upnp.device.utcnow",
            return_value=now + timedelta(seconds=DEFAULT_SCAN_INTERVAL),
        ):
            mock_device: MockIgdDevice = coordinator.device._igd_device
            mock_device.traffic_data = {
                BYTES_RECEIVED: int(10240 * DEFAULT_SCAN_INTERVAL),
                BYTES_SENT: int(20480 * DEFAULT_SCAN_INTERVAL),
                PACKETS_RECEIVED: int(30 * DEFAULT_SCAN_INTERVAL),
                PACKETS_SENT: int(40 * DEFAULT_SCAN_INTERVAL),
            }
            async_fire_time_changed(hass, now + timedelta(seconds=DEFAULT_SCAN_INTERVAL))
            await hass.async_block_till_done()
    
            kib_s_received_state = hass.states.get("sensor.mock_name_kib_s_received")
            kib_s_sent_state = hass.states.get("sensor.mock_name_kib_s_sent")
            packets_s_received_state = hass.states.get(
                "sensor.mock_name_packets_s_received"
            )
            packets_s_sent_state = hass.states.get("sensor.mock_name_packets_s_sent")
            assert kib_s_received_state.state == "10.0"
            assert kib_s_sent_state.state == "20.0"
            assert packets_s_received_state.state == "30.0"
>           assert packets_s_sent_state.state == "40.0"
E           AssertionError: assert '39.9' == '40.0'
E             - 40.0
E             + 39.9
```

### What version of Home Assistant Core has the issue?

dev

### What was the last working version of Home Assistant Core?

n/a

### What type of installation are you running?

Home Assistant Core

### Integration causing the issue

upnp

### Link to integration documentation on our website

https://www.home-assistant.io/integrations/upnp

### Example YAML snippet

_No response_

### Anything in the logs that might be useful for us?

_No response_

### Additional information

https://github.com/home-assistant/core/runs/4899938152?check_suite_focus=true

--- a/tests/components/upnp/test_sensor.py
+++ b/tests/components/upnp/test_sensor.py
@@ -114,7 +114,7 @@ async def test_derived_upnp_sensors(
             "sensor.mock_name_packets_s_received"
         )
         packets_s_sent_state = hass.states.get("sensor.mock_name_packets_s_sent")
-        assert float(kib_s_received_state.state) == pytest.approx(10.0)
-        assert float(kib_s_sent_state.state) == pytest.approx(20.0)
-        assert float(packets_s_received_state.state) == pytest.approx(30.0)
-        assert float(packets_s_sent_state.state) == pytest.approx(40.0)
+        assert float(kib_s_received_state.state) == pytest.approx(10.0, rel=0.1)
+        assert float(kib_s_sent_state.state) == pytest.approx(20.0, rel=0.1)
+        assert float(packets_s_received_state.state) == pytest.approx(30.0, rel=0.1)
+        assert float(packets_s_sent_state.state) == pytest.approx(40.0, rel=0.1)

--- a/tests/components/upnp/test_sensor.py
+++ b/tests/components/upnp/test_sensor.py
@@ -3,6 +3,8 @@
 from datetime import timedelta
 from unittest.mock import patch
 
+import pytest
+
 from homeassistant.components.upnp import UpnpDataUpdateCoordinator
 from homeassistant.components.upnp.const import (
     BYTES_RECEIVED,
@@ -112,7 +114,7 @@ async def test_derived_upnp_sensors(
             "sensor.mock_name_packets_s_received"
         )
         packets_s_sent_state = hass.states.get("sensor.mock_name_packets_s_sent")
-        assert kib_s_received_state.state == "10.0"
-        assert kib_s_sent_state.state == "20.0"
-        assert packets_s_received_state.state == "30.0"
-        assert packets_s_sent_state.state == "40.0"
+        assert float(kib_s_received_state.state) == pytest.approx(10.0)
+        assert float(kib_s_sent_state.state) == pytest.approx(20.0)
+        assert float(packets_s_received_state.state) == pytest.approx(30.0)
+        assert float(packets_s_sent_state.state) == pytest.approx(40.0)

